sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py

```diff
@@ -88,17 +92,34 @@ def get_cached_access_token(self, scopes: Iterable[str], **kwargs: Any) -> Optio
         )
 
         cache = self._get_cache(**kwargs)
-        for token in cache.search(
-            TokenCache.CredentialType.ACCESS_TOKEN,
-            target=list(scopes),
-            query={"client_id": self._client_id, "realm": tenant},
-        ):
+        results = list(
+            cache.search(
+                TokenCache.CredentialType.ACCESS_TOKEN,
+                target=list(scopes),
+                query={"client_id": self._client_id, "realm": tenant},
+            )
+        )
+        _CACHE_LOGGER.info(f"Using cache with ID {id(self._cache)}, credential client ID: {id(self)}")
```

Comment:
Is there a reason why we have to log entries? Fewer is better. More terse is better.